plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/driver/DriverDescriptor.java

```diff
@@ -366,6 +369,7 @@ public DriverDescriptor(DataSourceProviderDescriptor providerDescriptor, IConfig
             CommonUtils.getBoolean(config.getAttribute(RegistryConstants.ATTR_PROPAGATE_DRIVER_PROPERTIES));
         this.licenseRequired = CommonUtils.getBoolean(config.getAttribute(RegistryConstants.ATTR_LICENSE_REQUIRED));
         this.supportsDistributedMode = CommonUtils.getBoolean(config.getAttribute(RegistryConstants.ATTR_SUPPORTS_DISTRIBUTED_MODE), true);
+        this.notAvailableDriver = CommonUtils.getBoolean(config.getAttribute(RegistryConstants.ATTR_NOT_AVAILABLE_DRIVER), false);
```

Comment:
I wouldn't say I like that it depends on the `deprecated` property. At this point, I'd turn it from an attribute to a separate tag inside the `driver` tag.

A deprecation example:
```xml
<driver ...>
    <notAvailable deprecated="true" reason="%deprecation.text"/>
</driver>
```

A general unavailability:
```xml
<driver ...>
    <notAvailable reason="%deprecation.text"/>
</driver>
```

Or, to simplify things, the `deprecated` attribute can be dropped entirely:
```xml
<driver ...>
    <notAvailable title="%customTitle" description="%customDescription" message="%descriptiveMessage"/>
</driver>
```

By default, the `title` and `description` can be omitted and replaced with a "not available" message. Then, for ODBC, you can manually specify a custom title and description.

Reply:
Ok